test/test_2d_akima.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright Matt Borland, 2021
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+#include <boost/math/interpolators/2d_akima.hpp>
+
+int main(void)
+{
+    // Data from table 1
+    std::vector<double> x {11.16, 24.20, 19.85, 10.35, 19.72,  0.00, 20.87, 19.99, 10.28,  4.51,  0.00, 16.70,  6.08, 25.00, 14.90,  
+                            0.00,  9.66,  5.22, 11.77, 15.10, 25.00, 25.00, 14.59, 15.20,  5.23,  2.14,  0.51, 25.00, 21.67,  3.31};
+    
+    std::vector<double> y { 1.24, 16.23, 10.72,  4.11,  1.39, 20.00, 20.00,  4.62, 15.16, 20.00,  4.48, 19.65,  4.58, 11.87,  3.12,
+                            0.00, 20.00, 14.66, 10.47, 17.19,  3.87,  0.00,  8.71,  0.00, 10.72, 15.03,  8.37, 20.00, 14.36,  0.13};
+    
+    std::vector<double> z {22.15,  2.83,  7.97, 22.33, 16.83, 34.60,  5.74, 14.72, 21.59, 15.61, 61.77,  6.31, 35.74,  4.40, 21.70,
+                           58.20,  4.73, 40.36, 13.62, 12.57,  8.74, 12.00, 14.81, 21.60, 26.50, 53.10, 49.43,  0.60,  5.52, 44.08};
+
+    std::vector<std::vector<double>> xyz {x, y, z};
```

Comment:
A vector of vectors is a pretty bad data structure in general; why not an `Eigen` matrix?

Reply:
Is it worth supporting a vector of vectors or array of arrays? With some abuse of `std::conditional` it can happen but I would need to build an adapter that would transverse an iterator across the vectors to mimic the behavior of say ublas iterators.

Reply:
I would say it is not worth supporting vector of vector. Array of arrays is more debatable because the storage is contiguous, but the syntax will be incompatible with all matrix classes, so I'd say don't try supporting array of arrays.

Reply:
Just lurking here.... but the thing to do is decide on a matrix *concept*, and then use traits classes etc to map disparate concrete types (uBlas, Eigen, contain of container etc) to something that your code can understand.

Back to lurking...

Reply:
@jzmaddock : Please lurk more!

But are you saying both a vector of vectors and an Eigen matrix can work at the same time?

Reply:
In the sense that one should be capable of being mapped to the other via an abstract interface/traits class.  I wouldn't see it as necessary to support *everything* from the get go though.  We should also try to have one eye on whatever might come along in the future, I'm somewhat surprised that there's still no std lib matrix/array support.

Reply:
The matrix has been a sore point since you can't do:

```
operator[](int64_t i, int64_t j) const;
```

and write `M[i,j]`. People I guess want this before the matrix class can become a real boy.